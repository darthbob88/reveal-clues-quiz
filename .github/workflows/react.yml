# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: Node.js CI

on:
  workflow_dispatch:
  push:
    branches: [master]
  pull_request:
    branches: [master]

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [16.x, 18.x]

    steps:
      - uses: actions/checkout@v2
      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node-version }}
      - run: npm ci
      - run: npm run build --if-present
      - run: npm test
  deploy:
    if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
    runs-on: ubuntu-22.04
    permissions:
      contents: write
    concurrency:
      group: ${{ github.workflow }}-${{ github.ref }}
    steps:
      - uses: actions/checkout@v3
      - name: Deploy react app to github pages
        uses: tanwanimohit/deploy-react-to-ghpages@v1.0.1
