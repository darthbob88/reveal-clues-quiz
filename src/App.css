.App {
  text-align: center;
  max-width: 60%;
  margin: 0 auto;
  height: 100vh;
  display: flex;
  flex-direction: column;
}
div.content {
  flex:1
}
details.description{
  margin: .5rem auto;
  text-align: justify;
}
.description summary{
  text-align: center;
  margin-bottom: .5rem;
}