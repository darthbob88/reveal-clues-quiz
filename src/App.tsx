import React, { useState } from "react";
import "./App.css";
import { defaultQuiz } from "./model/Quiz";
import { QuizComponent } from "./components/Quiz/QuizComponent";

function App() {
  const [score, setScore] = useState(0);
  const awardPoints = (questionScore: number) => {
    setScore(score + questionScore);
  };
  return (
    <div className="App">
      <header>Raymond's Quiz App</header>
      <article className="description">
        Each question in this quiz will involve a series of clues of
        hopefully-increasing obviousness. If you get the question right after
        the first clue, you get the maximum number of points, each additional
        clue you need reduces the value by 1 point, down to 1 point if you need
        all the clues. HOWEVER, you only get one chance to answer, and if you
        answer wrong, you get 0 points. So it's better to ask for another clue
        and get 3 points than get a question wrong and get 0.
      </article>
      <p>Current score: {score}</p>
      <QuizComponent quiz={defaultQuiz} awardPoints={awardPoints} />
    </div>
  );
}

export default App;
