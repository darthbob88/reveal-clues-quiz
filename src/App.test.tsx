import React from "react";
import { act, render } from "@testing-library/react";
import App from "./App";
import { MemoryRouter } from "react-router-dom";

test("renders the app properly", async () => {
  await act(async () => {
    const { getByText, container } = render(<MemoryRouter><App /></MemoryRouter>);
    const linkElement = getByText(/Raymond's Quiz App/i);
    expect(linkElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })
});
