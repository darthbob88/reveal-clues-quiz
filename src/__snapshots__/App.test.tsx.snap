// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app properly 1`] = `
<div>
  <div
    class="App"
  >
    <header>
      <h1>
        Raymond's Quiz App
      </h1>
      <a
        href="/reveal-clues-quiz"
      >
        Home
      </a>
      <details
        class="description"
      >
        <summary>
          How to Play
        </summary>
        Each question in this quiz will involve a series of increasingly-obvious clues. If you get the question right after the first clue, you get the maximum number of points (4 points if the question has 4 clues, 3 points if it has 3 clues, etc). Each additional clue you need reduces the value by 1 point, down to 1 point if you need all the clues. HOWEVER, you only get one chance to answer, and if you answer wrong, you get 0 points. So it's better to ask for another clue and get 3 points than get a question wrong and get 0 points.
        <br />
        <br />
        EG: If you know which US state is home to 
        <a
          href="https://www.atlasobscura.com/places/widow-jane-mine"
        >
          the Widow Jane Mine
        </a>
        , you get 4 points. If you need another clue, that this state is also home to the 
        <a
          href="https://www.atlasobscura.com/places/jello-gallery-museum"
        >
          JELL-O Gallery Museum
        </a>
         you will get only 3 points for a correct answer. If you need to hear that this state also has the 
        <a
          href="https://www.atlasobscura.com/places/amityville-horror-house"
        >
          Amityville Horror house
        </a>
        , you only get 2 points. If you can't get the answer until you hear that this state also has the 
        <a
          href="https://www.nps.gov/stli/index.htm"
        >
          Statue of Liberty Enlightening the World
        </a>
        , you only get 1 point.
        <br />
        On the other hand, if you answered wrongly based on any of those clues, you'd only get 0 points.
        <br />
        (The state is New York, FYI)
      </details>
      <details
        class="description"
      >
        <summary>
          Known Issues
        </summary>
        <ul>
          <li>
            Only got two quizzes. Looking to get some more, and to add a "Create your own quiz" feature, but for now, two hard-coded quizzes.
          </li>
          <li>
            Not accessible enough. I've made an effort to keep things high-contrast, properly keyboard-navigable, and use semantic HTML, but there's probably some other use cases I've missed.
          </li>
          <li>
            Due to programmer laziness, we are not particular about casing, but we only accept one spelling and phrasing. "the sex pistols" is correct, "Sex Pistols" is not.
          </li>
          <li>
            I am deeply unsatisfied with the layout and UI, but don't know enough to fix it properly. Feedback welcome, although I am not budging on the brutalism.
          </li>
        </ul>
      </details>
    </header>
    <div
      class="content"
    >
      <div
        class="question"
      >
        <h2>
          List of Quizzes
        </h2>
        <ul
          class="clues"
        >
          <p>
            Loading...
          </p>
        </ul>
      </div>
    </div>
    <footer
      class="NavFooter"
    >
      Â© 
      2022
       Copyright: Raymond Price
      <a
        href="/https://github.com/darthbob88/"
      >
        Github
      </a>
      <a
        href="/https://github.com/darthbob88/reveal-clues-quiz"
      >
        Github Repo for this Project
      </a>
      <a
        href="/https://twitter.com/darthbob88"
      >
        Twitter
      </a>
      <a
        href="/https://www.linkedin.com/in/raymond-price-10a3746b/"
      >
        Linkedin
      </a>
    </footer>
  </div>
</div>
`;
